File: aspnet-in-practice/08-09/VB/CoolMVCBlog/Models/PostMetadata.vb
Imports System.ComponentModel.DataAnnotations
Imports System.Data.Objects.DataClasses

Public Class PostMetadata
    <UIHint("Author")>
    Public Property AuthorId As Integer

    <UIHint("Categories")>
    Public Property Categories As EntityCollection(Of Category)

    <DataType(DataType.MultilineText)>
    Public Property Text As String
End Class

<MetadataType(GetType(PostMetadata))>
Partial Public Class Post

End Class

File: aspnet-in-practice/08-09/CS/CoolMVBlog/Models/PostMetadata.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.ComponentModel.DataAnnotations;
using System.Data.Objects.DataClasses;

namespace CoolMvcBlog.Models
{
    public class PostMetadata
    {
        [UIHint("Categories")]
        public EntityCollection<Category> Categories { get; set; }
        
        [UIHint("Author")]
        public int AuthorId { get; set; }

        [DataType(DataType.MultilineText)]
        public string Text { get; set; }
    }

    [MetadataType(typeof(PostMetadata))]
    public partial class Post
    { 

    }
}